Soma-Technologies-Inc/visitAfrica-backend
# #175494178 get notification when user edit trip

#### What does this PR do?
* Creation of notification on trip edit
* Get created notification
#### Description of Task to be completed?
* User should see a notification after editing his request
* I have these api working
`GET /api/notification`
#### How should this be manually tested?
- open your terminal and clone this repo by running `git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend` 
- run `git checkout ft-multicityrequest-175494074` to navigate to my branch
- run `git pull` to make sure you have updated code
- run `npm install` to install all needed packages
- run `npm run migrate` to create migrations for the database
- navigate to login route `POST /api/auth/login` and login to get token that will b eused as a header for multi city request
- use these body data `{
    "email": "loua@gmail.com",
    "password": "Loua@123"
}`
- you will get a jwt then
- navigate to multi city request route `POST /api/trip/multi-city` and use a body data like this `{
   "departureFrom": "1",
   "departureTo": "2",
   "startingDate": "2020-12-20",
   "returningDate": "2020-12-30",
   "multiCity": [
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      }
   ],
   "reason": "travel"
}`
- before making any request append a `authorisation` header which have value `bearer token` replace token with the jwt token you got from `login` route
- then navigate to `POST /api/trip/edit/:id` replace `:id` with the id of your created rip then edit trip using models same as this data examples `{
   "departureFrom": "1",
   "departureTo": "2",
   "startingDate": "2021-12-20",
   "returningDate": "2021-12-30"}`
- Then a notification will be self created and navigate to `GET /api/notification` to see the notification created by user
#### Any background context you want to provide?
#### Screenshots (if appropriate)
#### Pivottracker Stories
[#175494178](https://www.pivotaltracker.com/story/show/175494178)
